packages/angular/build/src/utils/index-file/auto-csp.ts

```diff
@@ -0,0 +1,252 @@
+    if (scriptContent) {
+      const loaderScript = createLoaderScript(scriptContent);
+      if (loaderScript) {
+        hashes.push(hashInlineScript(loaderScript));
+        rewriter.emitRaw(`<script>${loaderScript}</script>`);
+      }
+      scriptContent = undefined;
+    }
+    rewriter.emitStartTag(tag);
+  });
+
+  rewriter.on('text', (tag, html) => {
+    if (openedScriptTag && !getScriptAttributeValue(openedScriptTag, 'src')) {
+      hashes.push(hashInlineScript(html));
+    }
+    rewriter.emitText(tag);
+  });
+
+  rewriter.on('endTag', (tag, html) => {
+    if (tag.tagName === 'script') {
+      const src = getScriptAttributeValue(openedScriptTag!, 'src');
+      openedScriptTag = undefined;
+
+      if (src) {
+        return;
+      }
+    }
+
+    if (tag.tagName === 'body' || tag.tagName === 'html') {
+      // Write the loader script if a string of <script>s were the last opening tag of the document.
+      if (scriptContent) {
+        const loaderScript = createLoaderScript(scriptContent);
+        if (loaderScript) {
+          hashes.push(hashInlineScript(loaderScript));
+          rewriter.emitRaw(`<script>${loaderScript}</script>`);
+        }
+        scriptContent = undefined;
+      }
+    }
+    rewriter.emitEndTag(tag);
+  });
+
+  const rewritten = await transformedContent();
+
+  // Second pass to add the header
+  const secondPass = await htmlRewritingStream(rewritten);
+  secondPass.rewriter.on('endTag', (tag, _) => {
+    if (tag.tagName === 'head') {
+      // See what hashes we came up with!
+      secondPass.rewriter.emitRaw(
+        `<meta http-equiv="Content-Security-Policy" content="${getStrictCsp(hashes)}">`,
+      );
+    }
+    secondPass.rewriter.emitEndTag(tag);
+  });
+  return secondPass.transformedContent();
+}
+
+/**
+ * Returns a strict Content Security Policy for mittigating XSS.
+ * For more details read csp.withgoogle.com.
+ * If you modify this CSP, make sure it has not become trivially bypassable by
+ * checking the policy using csp-evaluator.withgoogle.com.
+ *
+ * @param hashes A list of sha-256 hashes of trusted inline scripts.
+ * @param enableTrustedTypes If Trusted Types should be enabled for scripts.
+ * @param enableBrowserFallbacks If fallbacks for older browsers should be
+ *   added. This is will not weaken the policy as modern browsers will ignore
+ *   the fallbacks.
+ * @param enableUnsafeEval If you cannot remove all uses of eval(), you can
+ *   still set a strict CSP, but you will have to use the 'unsafe-eval'
+ *   keyword which will make your policy slightly less secure.
+ */
+function getStrictCsp(
+  hashes?: string[],
+  // default CSP options
+  cspOptions: {
+    enableBrowserFallbacks?: boolean;
+    enableTrustedTypes?: boolean;
+    enableUnsafeEval?: boolean;
+  } = {
+    enableBrowserFallbacks: true,
+    enableTrustedTypes: false,
+    enableUnsafeEval: false,
+  },
+): string {
+  hashes = hashes || [];
+  let strictCspTemplate = {
```

Comment:
Consider: I usually recommend `Map<string, T>` over `Record<string, T>` for anything with dynamically assigned keys to mitigate potential prototype pollution issues.

All the keys are string literals, so I don't think it's too important here, up to you if it's worth the effort.

Reply:
I'm not quite sure I'm following-- this is for `strictCspTemplate`?